gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/iceberg/IcebergTableTest.java

```diff
@@ -333,4 +348,119 @@ protected static <T> void verifyAnyOrder(Collection<T> actual, Collection<T> exp
   }
+
+  @Test
+  public void testGetPartitionSpecificDataFiles() throws IOException {
+    TableIdentifier testTableId = TableIdentifier.of(dbName, "testTableForPartitionSpecificDataFiles");
+    Table testTable = catalog.createTable(testTableId, icebergSchema, icebergPartitionSpec);
+
+    List<String> paths = Arrays.asList(
+        "/path/tableName/data/id=1/file1.orc",
+        "/path/tableName/data/id=1/file3.orc",
+        "/path/tableName/data/id=1/file5.orc",
+        "/path/tableName/data/id=1/file4.orc",
+        "/path/tableName/data/id=1/file2.orc"
+    );
+    // Using the schema defined in start of this class
+    PartitionData partitionData = new PartitionData(icebergPartitionSpec.partitionType());
+    partitionData.set(0, "1");
+    List<PartitionData> partitionDataList = Collections.nCopies(paths.size(), partitionData);
+
+    addPartitionDataFiles(testTable, paths, partitionDataList);
+
+    IcebergTable icebergTable = new IcebergTable(testTableId,
+        catalog.newTableOps(testTableId),
+        catalogUri,
+        catalog.loadTable(testTableId));
+    // Using AlwaysTrue & AlwaysFalse Predicate to avoid mocking of predicate class
+    Predicate<StructLike> alwaysTruePredicate = partition -> true;
+    Predicate<StructLike> alwaysFalsePredicate = partition -> false;
+    Assert.assertEquals(icebergTable.getPartitionSpecificDataFiles(alwaysTruePredicate).size(), 5);
+    Assert.assertEquals(icebergTable.getPartitionSpecificDataFiles(alwaysFalsePredicate).size(), 0);
+
+    catalog.dropTable(testTableId);
+  }
+
+  @Test
+  public void testOverwritePartition() throws IOException {
+    TableIdentifier overwritePartitionTestTableId = TableIdentifier.of(dbName, "testTableForOverwritePartition");
+    Table testTable = catalog.createTable(overwritePartitionTestTableId, icebergSchema, icebergPartitionSpec);
+
+    List<String> paths = Arrays.asList(
+        "/path/tableName/data/id=1/file1.orc",
+        "/path/tableName/data/id=1/file2.orc"
+    );
+    // Using the schema defined in start of this class
+    PartitionData partitionData = new PartitionData(icebergPartitionSpec.partitionType());
+    partitionData.set(0, "1");
+    PartitionData partitionData2 = new PartitionData(icebergPartitionSpec.partitionType());
+    partitionData2.set(0, "1");
+    List<PartitionData> partitionDataList = Arrays.asList(partitionData, partitionData2);
+
+    addPartitionDataFiles(testTable, paths, partitionDataList);
+
+    IcebergTable icebergTable = new IcebergTable(overwritePartitionTestTableId,
+        catalog.newTableOps(overwritePartitionTestTableId),
+        catalogUri,
+        catalog.loadTable(overwritePartitionTestTableId));
+
+    verifyAnyOrder(paths, icebergTable.getCurrentSnapshotInfo().getAllDataFilePaths(), "data filepaths should match");
+
+    List<String> paths2 = Arrays.asList(
+        "/path/tableName/data/id=2/file3.orc",
+        "/path/tableName/data/id=2/file4.orc"
```

Comment:
I don't believe the path naming convention matters.  rather it comes down only to the `PartitionData`, correct?  if so, let's feed in test data that does NOT use that convention, to demonstrate it's irrelevant.

this also suggests to try the corner case of two partitions in the same directory.  when one is overwritten, let's ensure files for the other partition are untouched.